documentation/specs/src/interoperability/ethereum-bridge/transfers_to_ethereum.md

```diff
@@ -0,0 +1,135 @@
+their wrapped tokens, which the `#EthBridge` validity predicate will accept.
+
+Mints of a wrapped Namada token on Ethereum (including NAM, Namada's native token)
+will be represented by a data type like:
+```rust
+struct MintWrappedNam {
+    /// The Namada address owning the token
+    owner: NamadaAddress,
+    /// The address on Ethereum receiving the wrapped tokens
+    receiver: EthereumAddress,
+    /// The address of the token to be wrapped 
+    token: NamadaAddress,
+    /// The number of wrapped Namada tokens to mint on Ethereum
+    amount: Amount,
+}
+```
+
+If a user wishes to mint a wrapped Namada token on Ethereum, they must
+submit a transaction on Namada that:
+- stores `MintWrappedNam` on chain somewhere - TBD
+- sends the correct amount of Namada token to `#EthBridgeEscrow`
+
+## Batching
+
+Ethereum gas fees make it prohibitively expensive in many cases to submit
+the proof for a single transaction over the bridge. Instead, it is typically
+more economical to submit proofs of many transactions in bulk. This batching
+is described in this section.
+
+A pool of transaction from Namada to Ethereum will be kept by Namada. Every
+transaction to Ethereum that Namada validators approve will be added to this
+pool. We call this the _Bridge Pool_.
+
+The Bridge Pool should be thought of as a sort of mempool. When users who
+wish to move assets to Ethereum submit their transactions, they will pay some
+additional amount of NAM (of their choosing) as a way of covering the gas
+costs on Ethereum. Namada validators will hold these fees in a Bridge Pool
+Escrow.
+
+When a batch of transactions from the Bridge Pool is submitted by a user to
+Ethereum, Namada validators will receive notifications via their full nodes.
+They will then pay out the fees for each submitted transaction to the user who
+relayed these transactions (still in NAM). These will be paid out from the
+Bridge Pool Escrow.
+
+The idea is that users will only relay transactions from the Bridge Pool
+that make economic sense. This prevents DoS attacks by underpaying fees as
+well as obviating the need for Ethereum gas price oracles. It also means
+that transfers to Ethereum are not ordered, preventing other attack vectors.
+
+The Bridge Pool will be organized as a Merkle tree. Every time it is updated,
+the root of tree must be signed by a quorum of validators. When a user
+wishes to construct a batch of transactions to relay to Ethereum, they
+include the signed tree root and inclusion proofs for the subset of the pool
+they are relaying. This can be easily verified by the Ethereum smart contracts.
+
+If vote extensions are available, these are used to collect the signatures
+over the Merkle tree root. If they are not, these must be submitted as protocol
+transactions, introducing latency to the pool. A user wishing to relay will
+need to wait until a Merkle tree root is signed for a tree that
+includes all the transactions they wish to relay.
```

Comment:
How often should validators sign a new Merkle root? In the absence of vote extensions, I think we should have some regular schedule like every 500 blocks (~1 hour or so) validators should submit a signature over whatever the Merkle root is in that block (assuming the Merkle root has changed since the last time).

Reply:
This is a hard question. Not signing can halt the bridge since no one can generated proofs of their relayed events. But we do have rate limiting, I should look into this.